### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by TimmyChan99**
In this PR I:

- Create forms to perform the following functions:
   - Creates a Post on behalf of the current_user.
   - Create a comment on behalf of the current_user.
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -10,6 +10,17 @@
 	 </article>
```
Comment (SajjadAhmad14) [inline app/views/posts/show.html.erb:10]: There is an error when the user goes to the post's author show page. Kindly see the screenshot below.
post_error

File: app/views/comments/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<h2 class='headline'>Add Comment</h2>
+
+<% flash.each do |type, msg| %>
+  <div>
+    <%= msg %>
+  </div>
+<% end %>
+
+	<%= form_with model: @comment, url: user_post_comments_path, scope: :comment, class: 'the_form' do |f| %>
```
Comment (SajjadAhmad14) [inline app/views/comments/new.html.erb:9]: There is an error when the user tries to create a new comment and also when he tries to like the post. Kindly see the screenshot below
comment_error
.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Fatima, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/views/posts/show.html.erb (hunk 1 above)
Comment (TimmyChan99) [reply app/views/posts/show.html.erb:10]: **thank you @SajjadAhmad14 for the feedback, however, I've checked again and it's working on my side and it's not throwing any errors. I can add posts, comments, and likes and I can access the posts
here is the pages**

image
image
image

File: app/views/comments/new.html.erb (hunk 2 above)
Comment (TimmyChan99) [reply app/views/comments/new.html.erb:9]: @SajjadAhmad14 I've checked and it's working on my side, kindly can you check again.

**Review by descholar-ceo — APPROVED** [review]
Hi @TimmyChan99 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**